components/esp_lcd/include/esp_lcd_panel_vendor.h

```diff
@@ -67,6 +94,24 @@ esp_err_t esp_lcd_new_panel_nt35510(const esp_lcd_panel_io_handle_t io, const es
  *          - ESP_ERR_INVALID_ARG   if parameter is invalid
  *          - ESP_ERR_NO_MEM        if out of memory
  *          - ESP_OK                on success
+ *
+ * @note The default panel size is 128x64.
+ * @note Use esp_lcd_panel_ssd1306_config_t to set the correct size.
+ * Example usage:
+ * @code {c}
+ * 
+ * esp_lcd_panel_ssd1306_config_t ssd1306_config = {
+ *     .width = 128,
+ *     .height = 32
```

Comment:
seems like this example usage is out of date.

Reply:
But compared to the 

```c
typedef struct {
   /**
    * @brief Multiplex ratio: (0..63)
    *
    * Display's height minus one.
    */
   uint8_t mux_ratio;

   /**
    * @brief Enables alternative COM pin configuration.
    *
    * When unset then Sequential COM pin configuration is used.
    */
   bool com_pin_alt;

   /**
    * @brief COM Left/Right remap.
    */
   bool com_lr_remap;
} esp_lcd_panel_ssd1306_config_t;
```

This example version looks more intuitive.

Reply:
@suda-morris thanks for your comments, good catch on mismatched documentation comment. I had several iterations on the driver and forgot to update the docs.

First I wanted to provide simpler interface with just `width` and `height` to make it easier to use. However later I reconsidered this approach because:
* width seems to never be needed;
* I'm not sure that height is enough to reliably configure a display;
* ... which may be even more confusing to developers when they first try the driver with the default settings (128x64), it fails, then they find this config, set the correct values and it fails again because for example it's a display from another manufacturer. In that case it will be quite hard to override lcd commands sent to display.

I decided to switch to more "generic" (but less intuitive) API to provide developers more flexibility while still being high-level enough to not send raw lcd commands. Maybe I should have mentioned typical config parameters for the most popular display sizes (128x64, 128x32, 96x16 and maybe others) in the documentation header.

Or it could be a good idea to provide public constants like 
```
extern const esp_lcd_panel_ssd1306_config_t esp_lcd_panel_ssd1306_128x32 = {
  .mux_ratio = 0x2,
  .com_pin_alt = false,
  .com_lr_remap = false
}
```
... but I'm not quite sure about naming and overall coding style, that's my very first touch of the embedded area.

@suda-morris what do you think I should prefer:
* mention typical configs for different screens in the documentation comment;
* provide constants with configs for known display types;

I lean towards the constants path. If you agree could you please guide me on the coding style and naming I should use? I struggle with that so far.

Reply:
Hi @storoj I would prefer the first approach. As you also said, different manufacturers may require different settings. We can't provide the setting for all known display types as well, because that's the job of a BSP package. Ideally, the user will define his own BSP package like [these](https://components.espressif.com/components?q=Board+Support+Package), and in there, he can predefine the value of esp_lcd_panel_ssd1306_config_t according to his manufacture.